webfrontend/src/components/navbar/search-user-box.tsx

```diff
@@ -0,0 +1,120 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { triggerErrorMessage } from "../../utils/locals";
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
+import Popover from 'react-bootstrap/Popover';
+import { Link } from "react-router-dom";
+
+interface userProperties {
+  name: string,
+  profile_pic_url: string,
+  username: string,
+  profile_link: string
+}
+const SearchUserBox = () => {
+  const searchForm = useRef<HTMLFormElement>(null);
+  const searchField = useRef<HTMLInputElement>(null);
+  const searchResultBox = useRef<HTMLDivElement>(null);
+  const [username, setUserName] = useState<string>('');
+  const [results, setResults] = useState<Array<userProperties>>([]);
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [clickOutside, setIsClickOutside] = useState<boolean>(false); 
+
+  const searchUser = useCallback(() => {
+    setIsClickOutside(false);
+    if (username !== '' && !isSearching) {
+        setIsSearching(true);
+        if (username === '@') {
+            fetch('/user/search_username?username=' + username.slice(1))
+                .then(response => {
+                    setIsSearching(false);
+                    if (response.status !== 200) {
+                        triggerErrorMessage();
+                    } else {
+                        response.json().then(response => setResults(response.users))
+                    }
+                });
+        } else {
+            fetch('/user/search?username=' + username)
+                .then(response => {
+                    setIsSearching(false);
+                    if (response.status !== 200) {
+                        triggerErrorMessage();
+                    } else {
+                        
+                        response.json().then(response => {
+                          setResults(response.users) 
+                        })
+                    }
+                });
+        }
+    }
+  }, [isSearching, username]);
+
+  const submitSearchField = useCallback((event: React.SyntheticEvent<EventTarget>) => {
+    event.preventDefault();
+    searchUser();
+  }, [searchUser])
+
+  const handleChange = useCallback((value: string) => {
+    setUserName(value);
+    setIsSearching(false);
+    searchUser();
+  }, [searchUser])
```

Comment:
Should not be calling API on every change of the input. This causes a lot of bugs, similar to #13